src/main/java/tennisclub/dao/EventDaoImpl.java

```diff
@@ -0,0 +1,79 @@
+package tennisclub.dao;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import tennisclub.entity.Court;
+import tennisclub.entity.Event;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * @author Miroslav Demek
+ */
+@Repository
+@Transactional
+public class EventDaoImpl implements EventDao {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public void create(Event event) {
+        em.persist(event);
+    }
+
+    @Override
+    public Event findById(Long id) {
+        return em.find(Event.class, id);
+    }
+
+    @Override
+    public List<Event> findByCourt(Court court) {
+        return em.createQuery("select e from Event e where e.court = :court", Event.class)
+                .setParameter("court", court)
+                .getResultList();
+    }
+
+    @Override
+    public List<Event> findByStartTime(LocalDateTime startTime) {
+        return em.createQuery("select e from Event e where e.startTime = :startTime", Event.class)
+                .setParameter("startTime", startTime)
+                .getResultList();
+    }
+
+    @Override
+    public List<Event> findByEndTime(LocalDateTime endTime) {
+        return em.createQuery("select e from Event e where e.endTime = :endTime", Event.class)
+                .setParameter("endTime", endTime)
+                .getResultList();
+    }
+
+    @Override
+    public List<Event> findByTimeInterval(LocalDateTime from, LocalDateTime to) {
+        return em.createQuery("select e from Event e where e.endTime >= :from and e.startTime <= :to", Event.class)
```

Comment:
I might be stupid _(compared to genius Mira 😁)_ but can `e.startTime > e.endTime` happen _(for example, `e.endTime = from and e.startTime = to` )_? My brain says that it should be `e.startTime >= from and e.endTime <= to` but I might be mistaken 😱

Reply:
@Fuyunori Your solution would not fulfill Mira's javadoc:
_Retrieve all events that at least **partially** take place
during the specified time interval. The interval is inclusive._

E.g. `start < from < to < end` should pass

I believe Mira is correct however this is definitely something that should be properly proved by tests.

Reply:
The formula you suggest would work only for events that are completely contained in the interval I think. The way it is now it should also allow scenarios when the event ends or begins in the middle of the requested interval. It is up to debate if this is what we want.

And I don't think `e.startTime > e.endTime` can happen, I think that is something that will be enforced on higher layers. But that reminds me that maybe I should add a note into the javadoc that the precondition is that `from <= to`?

Reply:
@bigno78 I believe `e.startTime > e.endTime` could be verified by this DAO in methods create and update (or, maybe even a better solution, by a database constraint)